### PR TITLE
add voidStatement and getStatementIterator to xApiEnhancedClient

### PR DESCRIPTION
**Pull request by Selindek**

### CODE REVIEW

**Review by thomasturrell — COMMENTED** [review]
Should we auto configure the enhanced client?

File: xapi-client/src/main/java/dev/learning/xapi/client/XapiEnhancedClient.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+    @Override
+    public boolean hasNext() {
+      return statements.hasNext() || more != null;
+    }
+
+    @Override
+    public Statement next() {
+      if (!statements.hasNext()) {
+        if (more == null) {
```
Comment (thomasturrell) [inline xapi-client/src/main/java/dev/learning/xapi/client/XapiEnhancedClient.java:210]: this surprised me, I believe that it in a conferment LRS it should be ""

File: xapi-client/src/test/java/dev/learning/xapi/client/XapiEnhancedClientTests.java [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        .build();
+
+    // When Voiding Statement
+    client.voidStatement(attemptedStatement).block();
+
+    final var recordedRequest = mockWebServer.takeRequest();
+
+    // Then Body Is Expected
+    assertThat(recordedRequest.getBody().readUtf8(), is(
+        "{\"actor\":{\"objectType\":\"Agent\",\"name\":\"A N Other\",\"mbox\":\"mailto:another@example.com\"},\"verb\":{\"id\":\"http://adlnet.gov/expapi/verbs/voided\",\"display\":{\"und\":\"voided\"}},\"object\":{\"objectType\":\"StatementRef\",\"id\":\"175c9264-692f-4108-9b7d-0ba64bd59ac3\"}}"));
```
Comment (thomasturrell) [inline xapi-client/src/test/java/dev/learning/xapi/client/XapiEnhancedClientTests.java:273]: Add a test for an iterated transfer into a stream and limited. First response should have 2 results seciond response should have 1 result. there should only be one call.

File: xapi-client/src/test/java/dev/learning/xapi/client/XapiClientTests.java [hunk 3]
```diff
@@ -2243,6 +2246,269 @@ void whenGettingActivityProfilesWithoutSinceParameterThenPathIsExpected()
...
+    assertThat(iterator.hasNext(), is(false));
+
+  }
+
+  @Test
+  void givenEmptyResponseWhenGettingStatementIteratorThenHasNextIsFalse()
+      throws InterruptedException {
+
+    // Given Empty Response
+    // This response is technically invalid by the xApi specification, but we cannot trust the xApi
```
Comment (thomasturrell) [inline xapi-client/src/test/java/dev/learning/xapi/client/XapiClientTests.java:2443]:
```suggestion
    // This response is technically invalid by the xAPI specification, but we cannot assume conformance.
```

File: xapi-client/src/test/java/dev/learning/xapi/client/XapiClientTests.java [hunk 4]
```diff
@@ -2243,6 +2246,269 @@ void whenGettingActivityProfilesWithoutSinceParameterThenPathIsExpected()
...
+    // conformance of the commercial LRSs.
```
Comment (thomasturrell) [inline xapi-client/src/test/java/dev/learning/xapi/client/XapiClientTests.java:2444]:
```suggestion
    // conformance of the commercial LRSs.
```

Comment (Selindek) [reply xapi-client/src/test/java/dev/learning/xapi/client/XapiClientTests.java:2444]: those are same

**Pull request merged**